aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -203,22 +203,19 @@ impl VMChangeSet {
 
     /// Builds a new change set from the storage representation.
     ///
+    /// **WARNING**: this creates a write set that assumes delayed field optimization to be disabled.
+    /// this needs to be applied directly to storage, you cannot get appropriate reads from this in a
+    /// delayed field optimization enabled context.
```

Comment:
does it make sense to mention technical intuition / context, i.e. the fact that the values and IDs are not exchanged for delayed fields.
because reading the above accomplishes the main goal, of deterring incorrect use, but possibly whoever reads it may want to find out more to why that is (and to hint where to look).